lamindb/core/storage/_tiledbsoma.py

```diff
@@ -186,11 +192,15 @@ def save_tiledbsoma_experiment(
             **kwargs,
         )
 
-    return Artifact(
+    artifact = Artifact(
         storepath,
         key=key,
         description=description,
         run=run,
         revises=revises,
         _is_internal_call=True,
-    ).save()
+    )
+    artifact.n_observations = n_observations
+    artifact._accessor = "TileDB-SOMA"
```

Comment:
Can we please keep it simple here and use `tiledbsoma` instead?

Reply:
I'll make the change.

It's mostly for aesthetical reasons, but `TileDB-SOMA` also isn't valid Python code whereas `tiledbsoma` is.